controllers/rule_controller.go

```diff
@@ -41,13 +48,56 @@ func (r *RuleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	ctx := context.Background()
 	_ = r.Log.WithValues("rule", req.NamespacedName)
-	rulesList := &oathkeeperv1alpha1.RuleList{}
 
-	err := r.List(ctx, rulesList)
+	var rule oathkeeperv1alpha1.Rule
+	skipValidation := false
+
+	if err := r.Get(ctx, req.NamespacedName, &rule); err != nil {
+		if apierrs.IsNotFound(err) {
+			skipValidation = true
+		} else {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if !skipValidation {
+		if err := rule.ValidateWith(r.ValidationConfig); err != nil {
+			rule.Status.Validation = &oathkeeperv1alpha1.Validation{}
+			rule.Status.Validation.Valid = boolPtr(false)
+			e := err.Error()
+			rule.Status.Validation.Error = &e
+			r.Log.Info("validation error in Rule %s/%s: \"%s\"", rule.Namespace, rule.Name, e)
+			if err := r.Update(ctx, &rule); err != nil {
+				r.Log.Error(err, "unable to update Rule status")
+				return ctrl.Result{}, err
+			}
+			// continue, as validation can't be fixed by requeuing request and we still have to update the configmap
+		} else {
+			// rule valid - set the status
+			rule.Status.Validation = &oathkeeperv1alpha1.Validation{}
+			rule.Status.Validation.Valid = boolPtr(true)
+			if err := r.Update(ctx, &rule); err != nil {
+				r.Log.Error(err, "unable to update Rule status")
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	var rulesList oathkeeperv1alpha1.RuleList
+
```

Comment:
FOLLOW UP: IMO we should represent list of rules in configmap which divided by namespaces, so a key in configmap should be namespace and value a list of rules for that namespace.
Instead of listing all CRs from all namespaces we would list here only rules from the given namespace (based on what is in the Request object) and make an update of the key in the configmap